internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -167,7 +179,7 @@ func (c CmdClient) Push(uri string, tags ...string) (digest string, err error) {
 		}
 	}
 	buf := new(strings.Builder)
-	if err := c.runner.Run("docker", []string{"inspect", "--format", "'{{json (index .RepoDigests 0)}}'", uri}, exec.Stdout(buf)); err != nil {
+	if err := c.runner.Run("docker", []string{"inspect", "--format", "'{{json (index .RepoDigests 0)}}'", imageName(uri, tags[0])}, exec.Stdout(buf)); err != nil {
```

Comment:
can you add a comment explaining why we're indexing to 0? For example:
```
// The container image will have the same digest regardless of the associated tag.
// Pick the first tag and get the image's digest.
```

Reply:
Why do we add the first tag to the URI instead of previous `uri`?

Reply:
Previously even the sidecar images are also build with `latest` tag but we do not `push the latest tag` of sidecar images to ECR. In this way we get the Image digest with out referencing with the `tag`. i.e just using uri. 

After addressing some of the feedback from Parag in the previous PR #4547  that we do not need to build the sidecar images with `latest` tag. I changed the Implementation and it is fault from my end that I did not test the functionality of sidecar build.

With this change we know exactly reference the image with the help of `uri:tag`   to get image digest. Using the first tag because in case of main container `tgas[0] ` will always be latest` and for sidecar images `tags[0] will be <sidecar>-latest`.

Reply:
> Using the first tag because in case of main container tgas[0]  will always be latest and for sidecar images tags[0] will be <sidecarName>-latest.

I see gotcha. maybe add this to the comment as well? Sorry for not reading the description of the PR

```
For Main container we call  docker inspect --format '{{json (index .RepoDigests 0)}}' uri:latest
For Sidecar container images we call docker inspect --format '{{json (index .RepoDigests 0)}}' uri:<sidecarname>-latest
```